crates/wallet/src/wallet/coin_selection.rs

```diff
@@ -513,7 +518,9 @@ impl<Cs: CoinSelectionAlgorithm> CoinSelectionAlgorithm for BranchAndBoundCoinSe
             // remaining_amount can't be negative as that would mean the
             // selection wasn't successful
             // target_amount = amount_needed + (fee_amount - vin_fees)
-            let remaining_amount = (curr_value - signed_target_amount) as u64;
+            let remaining_amount = (curr_value - signed_target_amount)
+                .to_unsigned()
+                .expect("unsigned amount");
```

Comment:
This sounds like a bit stronger of a reason. same suggestion on L667
```suggestion
                .expect("remaining amount can't be negative");
```

Reply:
Fixed.